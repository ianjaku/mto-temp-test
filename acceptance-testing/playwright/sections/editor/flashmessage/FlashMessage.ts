import { TestSection } from "../../testsection";
import { TestSectionLocators } from "../testsectionlocators";
import { expect } from "@playwright/test";

export class FlashMessage extends TestSection {

    private locators = new FlashMessageLocators(this.context);

    async expectSuccessFlashMessage(flashMessageText?: string): Promise<void> {
        await expect(this.locators.successFlashMessage).toBeVisible();
        if (flashMessageText) {
            await expect(this.locators.successFlashMessage).toContainText(flashMessageText);
        }
    }

    async expectInfoFlashMessage(flashMessageText?: string): Promise<void> {
        await expect(this.locators.infoFlashMessage).toBeVisible();
        if (flashMessageText) {
            await expect(this.locators.infoFlashMessage).toContainText(flashMessageText);
        }
    }

}

export class FlashMessageLocators extends TestSectionLocators {

    successFlashMessage = this.page.locator(".flashmessage.success")
    infoFlashMessage = this.page.locator(".flashmessage.info")

}
