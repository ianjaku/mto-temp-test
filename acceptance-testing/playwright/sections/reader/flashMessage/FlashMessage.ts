import { TestSection } from "../../testsection";
import { TestSectionLocators } from "../../editor/testsectionlocators";
import { expect } from "@playwright/test";

export class FlashMessage extends TestSection {

    private locators = new FlashMessageLocators(this.context);

    async expectErrorFlashMessage(flashMessageText?: string): Promise<void> {
        await expect(this.locators.errorFlashMessage).toBeVisible();
        if (flashMessageText) {
            await expect(this.locators.errorFlashMessage).toContainText(flashMessageText);
        }
    }

    async expectInfoFlashMessage(flashMessageText?: string): Promise<void> {
        await expect(this.locators.infoFlashMessage).toBeVisible();
        if (flashMessageText) {
            await expect(this.locators.infoFlashMessage).toContainText(flashMessageText);
        }
    }
}

class FlashMessageLocators extends TestSectionLocators {

    errorFlashMessage = this.page.locator(".flashmessage.flashmessage-severity-error")
    infoFlashMessage = this.page.locator(".flashmessage.flashmessage-severity-info")

}
