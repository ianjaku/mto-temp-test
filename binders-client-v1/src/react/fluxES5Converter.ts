/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
