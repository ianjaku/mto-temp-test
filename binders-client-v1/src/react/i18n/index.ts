import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
