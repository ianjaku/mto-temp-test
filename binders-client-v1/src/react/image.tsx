/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
