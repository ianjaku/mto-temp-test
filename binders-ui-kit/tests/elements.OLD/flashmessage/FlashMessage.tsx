import * as React from "react";
import FlashMessage, { FlashMessageType } from "../../../src/elements/flashmessage";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const MESSAGE_SUCCESS = "A success message";
const MESSAGE_INFO = "An info message";
const MESSAGE_ERROR = "An error message";

let flashMessageDismissed = false;

const dismissFlashMessage = () => flashMessageDismissed = true;


const createSuccessFM = () => wrapWithTheme(<FlashMessage type={FlashMessageType.SUCCESS} message={MESSAGE_SUCCESS} open={true} />);
const createInfoFM = () => wrapWithTheme(<FlashMessage type={FlashMessageType.INFO} message={MESSAGE_INFO} open={true} />);
const createErrorFM = () => wrapWithTheme(<FlashMessage type={FlashMessageType.ERROR} message={MESSAGE_ERROR} open={true} onHide={dismissFlashMessage} />);

// snapshots
test("Success flash message (snapshot)", () => {
    const fm1 = create(createSuccessFM());
    const serialized = fm1.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Info flash message (snapshot)", () => {
    const fm2 = create(createInfoFM());
    const serialized = fm2.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Error flash message (snapshot)", () => {
    const fm3 = create(createErrorFM());
    const serialized = fm3.toJSON();
    expect(serialized).toMatchSnapshot();
});

// enzyme
test("Success flash message (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createSuccessFM());
    const flashMessage = fmMount.find(".flashmessage").first();
    const message = flashMessage.find(".message").first().text();
    expect(message).toEqual(MESSAGE_SUCCESS);
    expect(flashMessage.hasClass("success")).toEqual(true);
});

// enzyme
test("Info flash message (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createInfoFM());
    const flashMessage = fmMount.find(".flashmessage").first();
    const message = flashMessage.find(".message").first().text();
    expect(message).toEqual(MESSAGE_INFO);
    expect(flashMessage.hasClass("info")).toEqual(true);
});

// enzyme
test("Error flash message (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createErrorFM());
    const flashMessage = fmMount.find(".flashmessage").first();
    const message = flashMessage.find(".message").first().text();
    expect(message).toEqual(MESSAGE_ERROR);
    expect(flashMessage.hasClass("error")).toEqual(true);
});

test("Flash message hiding on clicking close (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createErrorFM());
    expect(flashMessageDismissed).toEqual(false);
    fmMount.find(".close").first().simulate("click");
    expect(flashMessageDismissed).toEqual(true);
});
