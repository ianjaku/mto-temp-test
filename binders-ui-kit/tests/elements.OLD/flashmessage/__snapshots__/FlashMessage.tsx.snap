// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error flash message (snapshot) 1`] = `
<div
  className="flashmessage open error"
>
  <div
    className="icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      error_outline
    </span>
  </div>
  <label
    className="message"
  >
    An error message
  </label>
  <label
    className="close"
    onClick={[Function]}
  >
    <span
      className="material-icons "
      color="inherit"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "inherit",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      close
    </span>
  </label>
</div>
`;

exports[`Info flash message (snapshot) 1`] = `
<div
  className="flashmessage open info"
>
  <div
    className="icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      info_outline
    </span>
  </div>
  <label
    className="message"
  >
    An info message
  </label>
  <label
    className="close"
    onClick={undefined}
  >
    <span
      className="material-icons "
      color="inherit"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "inherit",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      close
    </span>
  </label>
</div>
`;

exports[`Success flash message (snapshot) 1`] = `
<div
  className="flashmessage open success"
>
  <div
    className="icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      done
    </span>
  </div>
  <label
    className="message"
  >
    A success message
  </label>
  <label
    className="close"
    onClick={undefined}
  >
    <span
      className="material-icons "
      color="inherit"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "inherit",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      close
    </span>
  </label>
</div>
`;
